src/backend/distributed/utils/metadata_cache.c

```diff
@@ -232,6 +232,21 @@ LoadShardInterval(uint64 shardId)
 
 
 /*
+ + * ReplicationModel returns the replication model for the given relation. The function
+ + * errors out for tables that are not distributed.
+ + */
+char
+ReplicationModel(Oid relationId)
```

Comment:
If you're very doubtful about this, I'm not going to insist on :) 
I remember your concerns on such functions. But given that in the functions I call this function, there is no access to the cache, I thought this would simplify the code. But, to prevent future callers of this function, I'm switching to accessing the cache itself.